### PR TITLE
Signup

### PR DESCRIPTION
**Pull request by AbdullahChaudhry**
Unfinished work, creating PR as requested by @rinoma so we can merge into master and all be on the same page

### CODE REVIEW

File: src/models/encrypt.js [hunk 1]
```diff
@@ -0,0 +1,24 @@
+const bcrypt = require('bcrypt')
+
+const hashPassword = (plaintextPassword) => {
+  return new Promise((resolve, reject) => {
+    bcrypt.hash(plaintextPassword, 10, (err, hash) => {
+      if (err) reject(Error('error hashing password'))
+      else if (hash) resolve(hash)
```
Comment (BartBucknill) [inline src/models/encrypt.js]: What if there is no hash?

File: src/models/verifyCharity.js [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+    Area of Work: ${areaOfWork}
+    Charity Address: ${charityAddress}
+    Contact Name: ${contactName}
+    Contact Surname: ${contactSurname}
+    Contact Email: ${contactEmail}
+    Contact Number: ${contactNumber}
+
+    What would you like to do:
+    Click here to approve:
+    http://0.0.0.0:4000/validate/email=${contactEmail}&str=${randomstring}
```
Comment (BartBucknill) [inline src/models/verifyCharity.js]: Can we make the url specific to the environment the site is running in?

File: src/models/verifyCharity.js [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+    Click here to reject:
+    http://0.0.0.0:4000/reject/email=${contactEmail}&str=${randomstring}
```
Comment (BartBucknill) [inline src/models/verifyCharity.js]: Is it OK to send the contact email in the url, not in the body?

File: src/models/encrypt.js (hunk 1 above)
Comment (AbdullahChaudhry) [reply src/models/encrypt.js]: bcrypt will do the error checking for us, I will include an else statement with resolve(hash)

File: src/models/verifyCharity.js (hunk 3 above)
Comment (AbdullahChaudhry) [reply src/models/verifyCharity.js]: Done. Just pushed changes to the branch

**Pull request merged**